keps/sig-cli/kustomize-ordering.md

```diff
@@ -0,0 +1,131 @@
+---
+title: Kustomize Resource Ordering
+authors:
+ - "@pwittrock"
+owning-sig: sig-cli
+participating-sigs:
+ - sig-apps
+ - sig-api-machinery
+reviewers:
+ - "@apelisse"
+ - "@anguslees"
+approvers:
+ - "@monopole"
+editors:
+ - "@pwittrock"
+creation-date: 2019-03-1
+last-updated: 2019-03-1
+status: implementable
+see-also:
+replaces:
+superseded-by:
+ - n/a
+---
+
+# Kustomize Resource Ordering
+
+## Table of Contents
+* [Table of Contents](#table-of-contents)
+* [Summary](#summary)
+* [Motivation](#motivation)
+  * [Goals](#goals)
+  * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+  * [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+  * [Risks and Mitigations](#risks-and-mitigations)
+* [Graduation Criteria](#graduation-criteria)
+* [Implementation History](#implementation-history)
+* [Alternatives](#alternatives)
+
+[Tools for generating]: https://github.com/ekalinin/github-markdown-toc
+
+## Summary
+
+Kustomize orders Resource creation by sorting the Resources it emits based off their type.  While
+this works well for most types (e.g. create Namespaces before other things), it doesn't work
+in all cases and users will need the ability to break glass.
+
+See [kubernetes-sigs/kustomize#836] for an example.
+
+## Motivation
+
+Users may need direct control of the Resource create / update / delete ordering in cases were
+sorting by Resource Type is insufficient.
+
+### Goals
+
+- Provide the ability for users to break glass and override Kustomize's sort ordering.
```

Comment:
See, I didn't even realized that kubectl had to apply the resources in the order they are received. Kubectl could also parallelize the requests to make order "less important". A lot of this is still going to be flaky because it has to assume that the resources will be created "within time-limit" (not just in-order), since we're talking about an asynchronous system here.

Reply:
Did you look at the example in the motivation?